### PR TITLE
[Reviewer: Matt] Added test for continuous accumulator 

### PR DESCRIPTION
**Pull request by TygerBurning**
Renamed AccumulatorTable to the new name EventAccumulatorTable and added UT for ContinuousAccumulatorTable, including for crossover period cases and non-boundary aligned period starts.

### CODE REVIEW

File: src/test_snmp.cpp [hunk 1]
```diff
@@ -538,3 +539,78 @@ TEST_F(SNMPTest, SuccessFailCountByRequestTypeTable)
...
 }
+
+// Advance to the next start of interval - accurate to within the first
+// second.
+void jump_to_next_periodstart(uint32_t interval_ms)
+{
+  struct timespec now;
+  clock_gettime(CLOCK_REALTIME_COARSE, &now);
+  uint64_t ms_since_epoch = (now.tv_sec * 1000) + (now.tv_nsec / 1000000);
+  cwtest_advance_time_ms(interval_ms - (ms_since_epoch % interval_ms));
```
Comment (mirw) [inline src/test_snmp.cpp:554]: Worth commenting what you're doing?

File: src/test_snmp.cpp [hunk 2]
```diff
@@ -538,3 +539,78 @@ TEST_F(SNMPTest, SuccessFailCountByRequestTypeTable)
...
+  FILE* fd = popen("snmpwalk -v2c -On -c clearwater 127.0.0.1:16161 .1.2.2.1.2.2", "r");
+  char buf[1024];
+  fgets(buf, sizeof(buf), fd);
+  ASSERT_STREQ(".1.2.2.1.2.2 = Gauge32: 200\n", buf); // Current 5 minutes
+
+  // Add another value to the table and advance 89 seconds
+  tbl->accumulate(100);
+  cwtest_advance_time_ms(89000);
+
+  // This period has spent half the time at 200, and half at 100
```
Comment (mirw) [inline src/test_snmp.cpp:585]: Hasn't it spent 90/179 at 200 and 89/179 at 100?  Worth explaining why not.

File: src/test_snmp.cpp [hunk 3]
```diff
@@ -538,3 +539,78 @@ TEST_F(SNMPTest, SuccessFailCountByRequestTypeTable)
...
+  // So average value should be 150
+  fd = popen("snmpwalk -v2c -On -c clearwater 127.0.0.1:16161 .1.2.2.1.2.2", "r");
+  fgets(buf, sizeof(buf), fd);
+  ASSERT_STREQ(".1.2.2.1.2.2 = Gauge32: 150\n", buf); // Current 5 minutes
+
+  // Jump forward to the next 5 minutes (half way through it)
```
Comment (mirw) [inline src/test_snmp.cpp]: Jumping forward by 150s would take us to (179 + 150) % 300 = 29s into the next period, not half-way through it.

File: src/test_snmp.cpp [hunk 4]
```diff
@@ -538,3 +539,78 @@ TEST_F(SNMPTest, SuccessFailCountByRequestTypeTable)
...
+  ASSERT_STREQ(".1.2.2.1.4.2 = Gauge32: 100\n", buf); // Current 5 minutes
+
+  fd = popen("snmpwalk -v2c -On -c clearwater 127.0.0.1:16161 .1.2.2.1.5.2", "r");
+  fgets(buf, sizeof(buf), fd); // LWM
+  ASSERT_STREQ(".1.2.2.1.5.2 = Gauge32: 100\n", buf); // Current 5 minutes
+
+  // The previous 5 minutes should not have changed value
+  fd = popen("snmpwalk -v2c -On -c clearwater 127.0.0.1:16161 .1.2.2.1.2.3", "r");
+  fgets(buf, sizeof(buf), fd);
+  ASSERT_STREQ(".1.2.2.1.2.3 = Gauge32: 150\n", buf); // Current 5 minutes
```
Comment (mirw) [inline src/test_snmp.cpp]: Comment is wrong.

File: src/test_snmp.cpp [hunk 5]
```diff
@@ -538,3 +539,78 @@ TEST_F(SNMPTest, SuccessFailCountByRequestTypeTable)
...
+
```
Comment (mirw) [inline src/test_snmp.cpp:654]: Worth checking variance, LWM and HWM on previous period (or before incrementing) - we don't currently check that, for example, we've got LWM and HWM the right way round.

**Comment by mirw** [comment]
Looks good.  A few comments inline.

File: src/test_snmp.cpp (hunk 3 above)
Comment (TygerBurning) [inline src/test_snmp.cpp]: Recall we advanced time by 120s at the beginning to prove start of periods don't need to be at boundaries, but I've made this clearer anyway

**Comment by mirw** [comment]
Feel free to merge when you're ready.

**Pull request merged**